### PR TITLE
Servicebus : NetworkruleSet 

### PR DESCRIPTION
**Pull request by v-Ajnava**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- X ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [X] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 1]
```diff
@@ -4408,6 +4499,92 @@
         }
       },
       "description": "The result of the List migrationConfigurations operation."
+    },
+	"Subnet": {
```
Comment (dsgouda) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: nit: please fix indentation

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 2]
```diff
@@ -4408,6 +4499,92 @@
...
+      "properties": {
+        "ipMask": {
+          "type": "string",
+          "description": "IP Mask"
+        },
+        "action": {
+          "type": "string",
+          "description": "The IP Filter Action",
+          "enum": [
+            "Allow"
```
Comment (dsgouda) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: out of curiosity, do you really want a single valued enum?

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 2 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: for now, yes. the intention is to support Deny in future.

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 1 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: fixed

**Comment by dsgouda** [comment]
@jhendrixMSFT I'll take over this one.

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5375 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5375/commits/79edde29e3597c27778907c41c3e048256e48b5e

**Comment by dsgouda** [comment]
startbuild

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 3]
```diff
@@ -2993,6 +2993,97 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/networkRuleSets/default": {
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: >networkRuleSets [](start = 130, length = 15)

Is this a proxy or tracked resource ?

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 4]
```diff
@@ -4408,6 +4499,93 @@
...
+                "$ref": "#/definitions/NWRuleSetIpRules"
+              },
+              "description": "List of IpRules"
+            }
+          }
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: >Resource [](start = 33, length = 8)

Is this a proxy or tracked resource ?

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 5]
```diff
@@ -4408,6 +4499,93 @@
...
+        "subnet": {
+          "$ref": "#/definitions/Subnet",
+          "description": "Subnet properties"
+        },
+        "ignoreMissingVnetServiceEndpoint": {
+          "type": "boolean",
+          "description": "Value that indicates whether to ignore missing VNet Service Endpoint"
+        }
+      },
+      "description": "The response from the List namespace operation."
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: >The response from the List namespace operation [](start = 22, length = 46)

update ?

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 6]
```diff
@@ -4408,6 +4499,93 @@
...
+            "Allow"
+          ],
+          "x-ms-enum": {
+            "name": "NetworkRuleIPAction",
+            "modelAsString": true
+          },
+          "default": "Allow"
+        }
+      },
+      "description": "The response from the List namespace operation."
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: >The response from the List namespace operation." [](start = 22, length = 48)

update ?

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 7]
```diff
@@ -4408,6 +4499,93 @@
         }
       },
       "description": "The result of the List migrationConfigurations operation."
+    },
+    "Subnet": {
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: >" [](start = 4, length = 1)

"required" properties not specified for the new types

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/examples/NameSpaces/VirtualNetworkRule/SBNetworkRuleSetCreate.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
+{
+  "parameters": {
+    "namespaceName": "sdk-Namespace-6019",
+    "resourceGroupName": "ResourceGroup",
+    "api-version": "2017-04-01",
+    "subscriptionId": "Subscription",
+    "parameters": {
+      "properties": {
+        "defaultAction": "Deny",
+        "virtualNetworkRules": [
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/examples/NameSpaces/VirtualNetworkRule/SBNetworkRuleSetCreate.json]: >virtualNetworkRules [](start = 9, length = 19)

This is taking arrays.. PUT request MUST replace server-side representation of the resource.. should NOT merge.

Comment (sanjaiganesh) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/examples/NameSpaces/VirtualNetworkRule/SBNetworkRuleSetCreate.json]: you can consider PATCH support if support for partial updates (additions) are desired. 

---
In reply to: [265808072](https://github.com/Azure/azure-rest-api-specs/pull/5361#discussion_r265808072) [](ancestors = 265808072)

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json [hunk 9]
```diff
@@ -2993,6 +2993,97 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NetworkRuleSet"
+            },
+            "description": "The Namespace IpFilterRule."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (sanjaiganesh) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: >200 [](start = 11, length = 3)

Does this complete synchronously ? just double checking.

**Comment by sanjaiganesh** [comment]
>             "description": "Eventhub error response describing why the operation failed.",

ServiceBus ?

---
Refers to: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json:2986 in 530b1bf. [](commit_id = 530b1bf768af8bde3f78795429c083d446f3ec5b, deletion_comment = False)

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 4 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: its a proxy resource.

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 5 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: updated.

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 6 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: updated

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 7 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: added required property

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/examples/NameSpaces/VirtualNetworkRule/SBNetworkRuleSetCreate.json (hunk 8 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/examples/NameSpaces/VirtualNetworkRule/SBNetworkRuleSetCreate.json]: yes, its replaces the resource. we are considering the PATCH support in future releases.

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json (hunk 9 above)
Comment (v-Ajnava) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json]: yes.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**